packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -76,37 +72,44 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
         }
     }
 
-    protected async updateOutlineForEditor(editor: EditorWidget) {
+    protected async updateOutlineForEditor(editor: EditorWidget | undefined) {
         if (this.outlineViewManager.open) {
             const model = await this.getModel(editor);
             this.publish(model);
         }
     }
 
-    protected async getModel(editor: EditorWidget): Promise<monaco.editor.IModel> {
-        const reference = await this.textModelService.createModelReference(editor.editor.uri);
-        const model = reference.object.textEditorModel;
-        this.toDispose.dispose();
-        this.toDispose.push(model.onDidChangeContent(ev => {
-            this.publish(model);
-        }));
-        return model;
+    protected async getModel(editor: EditorWidget | undefined): Promise<monaco.editor.IModel | undefined> {
+        if (editor) {
+            const monacoEditor = get(editor);
+            const model = monacoEditor!.getControl().getModel();
+            this.toDispose.dispose();
+            this.toDispose.push(model.onDidChangeContent(ev => {
+                this.publish(model);
+            }));
+            return model;
+        }
     }
 
-    protected async publish(model: monaco.editor.IModel) {
-        const entries: SymbolInformation[] = [];
+    protected async publish(model: monaco.editor.IModel | undefined) {
```

Comment:
this should be two methods
1) `computeSymbolInformations(IModel): SymbolInformation[]`
2) `publish(SymbolInformation[])`